lib/cylc/cycling/iso8601.py

```diff
@@ -281,9 +281,64 @@ def _iso_interval_nonzero(interval_string):
         return bool(interval)
 
 
+class ISO8601Exclusions(object):
```

Comment:
The `IntegerExclusions` and `ISO8601Exclusions` classes look pretty much identical now. `cylc.cycling.loader` provides abstractions for points/sequences, would it be possible to use this to consolidate these two classes?